src/vm-repair/azext_vm_repair/_params.py

```diff
@@ -0,0 +1,31 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.parameters import get_resource_name_completion_list
+
+# pylint: disable=line-too-long
+
+def load_arguments(self, _):
+
+    # REUSABLE ARGUMENT DEFINITIONS
+    name_arg_type = CLIArgumentType(options_list=['--name', '-n'], metavar='NAME')
+    existing_vm_name = CLIArgumentType(overrides=name_arg_type,
+                                       configured_default='vm',
+                                       help="The name of the Virtual Machine. You can configure the default using `az configure --defaults vm=<name>`",
+                                       completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines'), id_part='name')
+
+    with self.argument_context('vm repair') as c:
+        c.argument('vm_name', existing_vm_name)
+
+    with self.argument_context('vm repair swap-disk') as c:
+        c.argument('rescue_username', help='Admin username for rescue VM.')
+        c.argument('rescue_password', help='Admin password for the rescue VM.')
+
+    with self.argument_context('vm repair restore-swap') as c:
+        c.argument('rescue_vm_id', help='Rescue VM resource id.')
```

Comment:
Parameters ending with `-id` refer to GUIDs. Since this is an ARM resource, it should be `--rescue-vm` and should accept a name or ID.

Reply:
The rescue VM is within the difference resource group from the target faulty vm. So only name would not work. What should I do here?

Reply:
Never mind the previous comment. Actually now that I think of it, because rescue VM lives in another resource group, the group and name has to be both specified and ID is the best way to do it. This is because the rescue-vm name might be the same in different resource groups. Please advise. @tjprescott

Reply:
Even if you only accept an ARM ID, you need to drop the `-id` suffix.

Reply:
Wouldn't that confuse the user thinking that they could enter the rescue vm name?

Reply:
I wonder what @tjprescott would respond with. But in the worst case, you can clarify this in the help text and with examples.

Reply:
So we should burden the users to look through the help text and examples rather than being clear on the parameter naming?